packages/openapi-generator/src/parser/document.ts

```diff
@@ -46,7 +47,8 @@ export function parseSchemas(
       ...refs.getSchemaNaming(`#/components/schemas/${name}`),
       schema: parseSchema(schema, refs, options),
       description: refs.resolveObject(schema).description,
-      schemaProperties: parseSchemaProperties(schema)
+      schemaProperties: parseSchemaProperties(schema),
+      nullable: (!isReferenceObject(schema) && schema.nullable) ?? false
```

Comment:
[q] can a reference object not be nullable or is this a type issue again, because the reference type does not allow nullable? in that case I would prefer something like you did for the required properties: `'nullable' in schema && schema.nullable...`.

Reply:
yes, it's because refObj does not have this property. I rewrote it to:

`'nullable' in schema && schema.nullable ? true : false`